staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_test.go

```diff
@@ -2186,3 +2189,130 @@ func TestEnsureBackendPoolDeleted(t *testing.T) {
 		assert.Equal(t, test.expectedErr, err != nil, test.description+", but an error occurs")
 	}
 }
+
+func TestEnsureBackendPoolDeletedConcurrently(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	ss, err := newTestScaleSet(ctrl)
+	assert.NoError(t, err)
+
+	backendAddressPools := &[]network.BackendAddressPool{
+		{
+			ID: to.StringPtr(testLBBackendpoolID0),
+			BackendAddressPoolPropertiesFormat: &network.BackendAddressPoolPropertiesFormat{
+				BackendIPConfigurations: &[]network.InterfaceIPConfiguration{
+					{
+						Name: to.StringPtr("ip-1"),
+						ID:   to.StringPtr("/subscriptions/sub/resourceGroups/rg/providers/Microsoft.Compute/virtualMachineScaleSets/vmss/virtualMachines/0/networkInterfaces/nic"),
+					},
+				},
+			},
+		},
+		{
+			ID: to.StringPtr(testLBBackendpoolID1),
+			BackendAddressPoolPropertiesFormat: &network.BackendAddressPoolPropertiesFormat{
+				BackendIPConfigurations: &[]network.InterfaceIPConfiguration{
+					{
+						Name: to.StringPtr("ip-1"),
+						ID:   to.StringPtr("/subscriptions/sub/resourceGroups/rg/providers/Microsoft.Compute/virtualMachineScaleSets/vmss/virtualMachines/0/networkInterfaces/nic"),
+					},
+				},
+			},
+		},
+		{
+			// this would fail
+			ID: to.StringPtr(testLBBackendpoolID2),
+			BackendAddressPoolPropertiesFormat: &network.BackendAddressPoolPropertiesFormat{
+				BackendIPConfigurations: &[]network.InterfaceIPConfiguration{
+					{
+						Name: to.StringPtr("ip-1"),
+						ID:   to.StringPtr("/subscriptions/sub/resourceGroups/rg1/providers/Microsoft.Compute/virtualMachineScaleSets/vmss/virtualMachines/0/networkInterfaces/nic"),
+					},
+				},
+			},
+		},
+	}
+	expectedVMSS := compute.VirtualMachineScaleSet{
```

Comment:
any possible to add some helper funtions for generating VirtualMachineScaleSet and VirtualMachineScaleSetVM? They could be reused in other test cases